File: fsharp-functions/OpenAPI/src/model/MissingArgumentError.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open Newtonsoft.Json
open OpenAPI.Model.AnyType
open OpenAPI.Model.CodedErrorBase
open OpenAPI.Model.MissingArgumentErrorAllOf

module MissingArgumentError = 

  //#region MissingArgumentError

  [<CLIMutable>]
  type MissingArgumentError = {
    [<JsonProperty(PropertyName = "result")>]
    Result : AnyType;
    [<JsonProperty(PropertyName = "msg")>]
    Msg : AnyType;
    [<JsonProperty(PropertyName = "code")>]
    Code : AnyType;
    [<JsonProperty(PropertyName = "var_name")>]
    VarName : string;
  }
  
  //#endregion
  

File: scala-finch/src/main/scala/org/openapitools/models/MissingArgumentError.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import org.openapitools.models.AnyType
import org.openapitools.models.CodedErrorBase
import org.openapitools.models.MissingArgumentErrorAllOf

/**
 * 
 * @param result 
 * @param msg 
 * @param code 
 * @param varUnderscorename It contains the information about the missing parameter. 
 */
case class MissingArgumentError(result: AnyType,
                msg: AnyType,
                code: Option[AnyType],
                varUnderscorename: Option[String]
                )

object MissingArgumentError {
    /**
     * Creates the codec for converting MissingArgumentError from and to JSON.
     */
    implicit val decoder: Decoder[MissingArgumentError] = deriveDecoder
    implicit val encoder: ObjectEncoder[MissingArgumentError] = deriveEncoder
}
